packages/kbn-apm-synthtrace/src/lib/apm/client/apm_synthtrace_kibana_client.ts

```diff
@@ -32,24 +32,42 @@ export class ApmSynthtraceKibanaClient {
 
   async fetchLatestApmPackageVersion() {
     this.logger.debug(`Fetching latest APM package version`);
-    const url = `${this.getFleetApmPackagePath()}?prerelease=true`;
-    const response = await fetch(url, {
-      method: 'GET',
-      headers: kibanaHeaders(),
-      agent: getFetchAgent(url),
-    });
 
-    const responseJson = await response.json();
+    const fetchPackageVersion = async ({ prerelease }: { prerelease: boolean }) => {
+      const url = `${this.getFleetApmPackagePath()}?prerelease=${prerelease}`;
+      this.logger.debug(`Fetching from URL: ${url}`);
 
-    if (response.status !== 200) {
-      throw new Error(
-        `Failed to fetch latest APM package version, received HTTP ${response.status} and message: ${responseJson.message}`
-      );
-    }
+      const response = await fetch(url, {
+        method: 'GET',
+        headers: kibanaHeaders(),
+        agent: getFetchAgent(url),
+      });
+
+      const responseJson = await response.json();
 
-    const { latestVersion } = responseJson.item;
+      if (response.status !== 200) {
+        throw new Error(
+          `Failed to fetch APM package version, received HTTP ${response.status} and message: ${responseJson.message}`
+        );
+      }
+
+      return responseJson.item.latestVersion as string;
+    };
 
-    return latestVersion as string;
+    try {
+      // First attempt fetch latest Prerelease version (prerelease=true)
+      return await fetchPackageVersion({ prerelease: true });
+    } catch (error) {
+      this.logger.debug(
+        'Fetching latestes prerelease version failed, retrying with latest GA version'
+      );
+      // Retry with fetch latest GA version (prerelease=false)
+      const retryResult = await fetchPackageVersion({ prerelease: false }).catch((retryError) => {
+        throw retryError;
+      });
```

Comment:
I think you can drop the comments here too. The code is pretty obvious. 

Also, do you still need to add the `catch` block on the retry?

Reply:
I added to avoid having an unhandled promise rejection. Do you think it's not necessary?
I look where we call `fetchLatestApmPackageVersion()` and I'm not sure that it will be handled there

Reply:
Can you try it out by juts throwing a new error when `prerelease: false` and remove the catch to see what's going to happen?